### PR TITLE
Various form improvements

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

File: examples/lit/src/composite-form/composite-form.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      'sl-button-bar': ButtonBar,
+      'sl-checkbox': Checkbox,
+      'sl-checkbox-group': CheckboxGroup,
+      'sl-form': Form,
+      'sl-form-field': FormField,
+      'sl-form-validation-errors': FormValidationErrors,
+      'sl-radio': Radio,
+      'sl-radio-group': RadioGroup,
+      'sl-text-field': TextField,
+      'sl-textarea': Textarea
```
Comment (anna-lach) [inline examples/lit/src/composite-form/composite-form.ts:26]: Are we going to change component name from `sl-textarea` to `sl-text-area`?

Comment (anna-lach) [reply examples/lit/src/composite-form/composite-form.ts:26]: I mean in general :) not only here

File: packages/components/form/src/form-validation-errors.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    'sl-form-validation-errors': FormValidationErrors;
+  }
+}
+
+@localized()
+export class FormValidationErrors extends ScopedElementsMixin(LitElement) {
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-inline-message': InlineMessage
```
Comment (anna-lach) [inline packages/components/form/src/form-validation-errors.ts:20]: Maybe `sl-icon` is missing?

image

File: packages/components/form/src/form-field.ts [hunk 3]
```diff
@@ -55,12 +55,12 @@ export class FormField extends ScopedElementsMixin(LitElement) {
...
   control?: HTMLElement & FormControl;
 
-  /**
-   * The validation message that will be displayed when the field is in an invalid state.
-   * @private
-   */
+  /** @internal The message that will be displayed when the field is in an invalid state. */
   @state() error?: string;
 
+  /** @internals A record of error messages for all controls. */
```
Comment (anna-lach) [inline packages/components/form/src/form-field.ts:61]: typo? 
```suggestion
  /** @internal A record of error messages for all controls. */
```

**Review by anna-lach — APPROVED** [review]

File: examples/lit/src/composite-form/composite-form.ts (hunk 1 above)
Comment (jpzwarte) [reply examples/lit/src/composite-form/composite-form.ts:26]: Yes, want to do that after this :)

File: packages/components/form/src/form-validation-errors.ts (hunk 2 above)
Comment (jpzwarte) [reply packages/components/form/src/form-validation-errors.ts:20]: I have the inline-message in another PR, so i don't really want to check it until then.

**Pull request merged**